src/math.ts

```diff
@@ -147,13 +147,29 @@ export const centerPoint = (a: Point, b: Point): Point => {
 // to be considered a loop
 export const isPathALoop = (
   points: ExcalidrawLinearElement["points"],
+  /** supply if you want the loop detection to account for current zoom */
+  zoomValue: Zoom["value"] = 1 as NormalizedZoomValue,
 ): boolean => {
   if (points.length >= 3) {
     const [firstPoint, lastPoint] = [points[0], points[points.length - 1]];
-    return (
-      distance2d(firstPoint[0], firstPoint[1], lastPoint[0], lastPoint[1]) <=
-      LINE_CONFIRM_THRESHOLD
+    const distance = distance2d(
+      firstPoint[0],
+      firstPoint[1],
+      lastPoint[0],
+      lastPoint[1],
     );
+
+    // Adjusting LINE_CONFIRM_THRESHOLD to current zoom so that when zoomed in
+    // really close we make the threshold smaller, and vice versa.
+    // The formulas are a result of curve-fitting target values.
+    const k = 3 - 2 * Math.pow(zoomValue, 0.05);
+    const threshold =
+      (LINE_CONFIRM_THRESHOLD / Math.log1p(zoomValue * k)) *
+      // multiplying by log reciprocal will cancel it out when zoomValue
+      // is 100%, so that threshold always results in LINE_CONFIRM_THRESHOLD
+      (Math.log1p(1 * k) / 1);
```

Comment:
can we split this into variables and add the comment there? It's a bit weird looking expression with long comment in between.

Reply:
I think it's fine, plus the interpreter may not do optimizations if split into two variables that could in theory result in floating point math issues (i.e. it would not cancel the two vars out) — probably a stretch and not a big deal, but styling is also not a big deal. Prettier at times does much worse things.

Reply:
can we at least change `k` to something more meaningful even though it has 3 lines of comment?

Reply:
I'd rather not call it anything because it's really arbitrary. Giving it a name (which?) would project meaning which may not be there. It's customary to use single-letter variables in math. I'll nevertheless modify the comment so it's more obvious what's going on.